### PR TITLE
[FIX] web: use read to update record values after resequence

### PR DESCRIPTION
**Pull request by len-foss**
Commit:
https://github.com/odoo/odoo/commit/818c18e55d0718286ff5bf332186a10f4d7a58ef
Updated the resequence logic in a way that was almost falser than before.
The added test however did work by coincidence, as index values were equal to
the sequence field values.
To be sure that we synchronize with what happens on the server, we do a read
after the resequence.
Additionnally, we take into account the result of the server resequence return;
if it is false, it means no resequencing happened, so we should not do a read.

opw 1867049

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by len-odoo** [comment]
@aab-odoo 
in fact wasResequenced already takes care of knowing if resequenced happened or not, so there is nothing more to do to fix the kanban warnings.

**Comment by aab-odoo** [comment]
@nim-odoo As far as i know, the editable list uses it as well

**Comment by nim-odoo** [comment]
@aab-odoo https://drive.google.com/file/d/1KAIY065i-E0SXuskBqeH9DJt9Bo6IL5C/view?usp=sharing

File: addons/web/static/src/js/views/basic/basic_model.js [hunk 1]
```diff
@@ -893,27 +893,40 @@ var BasicModel = AbstractModel.extend({
...
+                            record.data[field] = records[i][field];
+                        })
                     }
-                });
-                self._updateParentResIDs(data);
-                return parentID;
+                    data.data = _.sortBy(data.data, function (d) {
+                        return self.localData[d].data[field];
+                    });
+                    data.res_ids = [];
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/basic/basic_model.js]: this seems wrong (data.res_ids may contain more entries that data.data)

File: addons/web/static/tests/views/list_tests.js [hunk 2]
```diff
@@ -3129,14 +3131,15 @@ QUnit.module('Views', {
                     '<field name="amount" widget="float" digits="[5,0]"/>' +
                   '</tree>',
             mockRPC: function (route, args) {
+                var self = this;
+                var _super = this._super;
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list_tests.js]: none of these a used

File: addons/web/static/tests/views/list_tests.js [hunk 3]
```diff
@@ -3266,13 +3269,17 @@ QUnit.module('Views', {
                   '</tree>',
             mockRPC: function (route, args) {
                 if (route === '/web/dataset/resequence') {
+                    var self = this;
+                    var _super = this._super;
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list_tests.js]: var _super = this._super.bind(this);

File: addons/web/static/tests/views/list_tests.js [hunk 4]
```diff
@@ -3266,13 +3269,17 @@ QUnit.module('Views', {
...
                     assert.strictEqual(args.offset, 1,
                         "should write the sequence starting from the lowest current one");
                     assert.strictEqual(args.field, 'int_field',
                         "should write the right field as sequence");
                     assert.deepEqual(args.ids, [4, 2, 3],
                         "should write the sequence in correct order");
-                    return $.when(def);
+                    return $.when(def).then(function () {
+                        return _super.call(self, route, args);
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list_tests.js]: return _super(route, args);

File: addons/web/static/tests/views/list_tests.js [hunk 5]
```diff
@@ -3266,13 +3269,17 @@ QUnit.module('Views', {
                   '</tree>',
             mockRPC: function (route, args) {
                 if (route === '/web/dataset/resequence') {
+                    var self = this;
```
Comment (aab-odoo) [inline addons/web/static/tests/views/list_tests.js]: no need

File: addons/web/static/src/js/views/basic/basic_model.js [hunk 6]
```diff
@@ -893,27 +893,51 @@ var BasicModel = AbstractModel.extend({
...
+                            });
+                        }
                     }
-                });
-                self._updateParentResIDs(data);
-                return parentID;
+                    data.res_ids = [];
+                    _.each(data.data, function (d) {
+                        var dataPoint = self.localData[d];
+                        if (dataPoint.type === 'record') {
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/basic/basic_model.js]: dataPoint.type is dataType, i think

File: addons/web/static/src/js/views/basic/basic_model.js [hunk 7]
```diff
@@ -893,27 +893,51 @@ var BasicModel = AbstractModel.extend({
...
+                var field = params.field ? params.field : 'sequence';
+
+                return self._rpc({
+                    model: modelName,
+                    method: 'read',
+                    args: [resIDs, [field]],
+                }).then(function (records) {
+                    if (data.data.length) {
+                        var dataType = self.localData[data.data[0]].type;
+                        if (dataType === 'record'){
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/basic/basic_model.js]: ) {

File: addons/web/static/src/js/views/basic/basic_model.js [hunk 8]
```diff
@@ -893,27 +893,51 @@ var BasicModel = AbstractModel.extend({
...
+                                var inRecords = _.findWhere(records, {id: recordData.id});
+                                if (inRecords) {
+                                    recordData[field] = inRecords[field];
+                                }
+                            });
+                            data.data = _.sortBy(data.data, function (d) {
+                                return self.localData[d].data[field];
+                            });
+                        }
+                        if (dataType === 'list'){
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/basic/basic_model.js]: ) {

Comment (len-odoo) [reply addons/web/static/src/js/views/basic/basic_model.js]: Fixed these pesky whitespaces.

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**